lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -454,18 +480,19 @@ void PatchEntryPointMutate::fixupUserDataUses(Module &module) {
       Value *replacementVal = nullptr;
       if (userDataUsage->pushConstSpill) {
         // At least one use of the push constant pointer remains.
-        // TODO Shader compilation: If there is no user data layout, we need a reloc here.
-        assert(!m_pipelineState->getUserDataNodes().empty() && "Spill with no user data layout not implemented yet");
         const ResourceNode *node = m_pipelineState->findSingleRootResourceNode(ResourceNodeType::PushConst);
         Value *byteOffset = nullptr;
+        builder.SetInsertPoint(spillTable->getNextNode());
         if (node) {
           byteOffset = builder.getInt32(node->offsetInDwords * 4);
           // Ensure we mark spill table usage.
           m_pipelineState->getPalMetadata()->setUserDataSpillUsage(node->offsetInDwords);
-        } else {
+        } else if (!m_pipelineState->isUnlinked()) {
           byteOffset = UndefValue::get(builder.getInt32Ty());
+        } else {
+          // Unlinked shader compilation: Use a reloc.
+          byteOffset = builder.CreateRelocationConstant("pushconst");
```

Comment:
Could you please add documentation on what relocation constants are used and what they mean?

Reply:
Yes. That's probably best in the new header file I'm proposing above (which can also define const strings for ones like this).